stub_manager/src/lib/args_validator.rs

```diff
@@ -0,0 +1,42 @@
+use clap::{App, Arg};
+
+pub fn parse_and_validate() -> (String, String) {
+    let matches = App::new("stub_manager")
+        .about("manage locally running stubs")
+        .version("1.0")
+        .author("Ninad Mhatre")
+        .arg(
+            Arg::with_name("action")
+                .help("action can be <start|stop|status>")
+                .index(1)
+                .required(true)
+                .validator(validate_action),
+        )
+        .arg(
+            Arg::with_name("process")
+                .help("which stub process to manage? <ls|pk|all>")
+                .index(2)
+                .required(true)
+                .validator(validate_process),
+        )
+        .get_matches();
+
+    let action = matches.value_of("action").unwrap();
+    let process = matches.value_of("process").unwrap();
+
+    (action.to_string(), process.to_string())
+}
+
+fn validate_action(val: String) -> Result<(), String> {
```

Comment:
It's idiomatic to take `&str` arguments instead of owned types. That's because `&String` can be cast to `&str` at zero cost, but not the other way.

`String` is useful for storing it in `struct`s, and when you need to append to it in-place.